### ISSUE TITLE
Repeated Invalid HTTP Response Code 429 for Hangouts plugin

### ISSUE THREAD

**Issue by nekromoff** [EionRobb/purple-hangouts#222]
Reconnecting does not help. Sometimes deleting all certificates in `/home/xxxx/.purple/certificates/x509/tls_peers/` helps, other times it does not. Sometimes closing and restarting Pidgin helps, sometimes does not. No clear solution.

**Comment by sneakyjoeru** [comment]
Started happening to me today on windows version

**Comment by chrissharp123** [comment]
This issue just started for me this morning too. Using the Fedora repo version.

**Comment by sneakyjoeru** [comment]
It got fixed for me after a few days and account removal and reconnection. Can't tell what exactly helped tho.

**Comment by Otterverse** [comment]
I found it may be an issue with the NSS library for SSL not supporting some of the options that the google api server is using. It's intermittant because google has a lot of servers and I believe some are perhaps using different TLS versions than others. It's quasi-random which one you get on connect.

That said, I compiled pidgin using GNUTLS instead, and it worked fine, after repeatedly failing.

**Comment by Menelkir** [comment]
I've tried everything described here and none of then worked for me, any other suggestions? 
Strange that it works fine on my notebook and doesn't work on my desktop.

**Comment by rrthomas** [comment]
Thanks very much for the fix, @Otterverse!

**Comment by Annihilannic** [comment]
Ran into this one today under Windows, is any chance of a new release of the Windows installer with GNUTLS to get around this, please?

**Comment by EionRobb** [comment]
@Annihilannic no, Pidgin on Windows uses NSS and doesn't provide a gnutls option (not up to the plugin)

**Comment by Annihilannic** [comment]
Okay thanks for the clarification.  I think today's issue was a wider Hangouts problem, I noticed the official Hangouts Chrome app also had connectivity issues.

**Comment by nekromoff** [comment]
Still happening and not much changes, happens randomly and then randomly
connects again. Restarts or deleting certificates do not seem to help
anymore.

The issue appears when switching IP addresses (e.g. enabling VPN etc.) and
then goes away after random time.

**Comment by ctulloch137** [comment]
Anyone mind sharing a windows build compiled with gnuTLS?

**Comment by Industry** [comment]
This has been open for a few months now. Any update for Windows users?

**Comment by sneakyjoeru** [comment]
Started happening very frequently again. HAve no idea how to fix it this time.

**Comment by BoFiS** [comment]
Definitely has something to do with IP address changing or after a disconnection.  Our internet at work cut out, and when it came back I'm getting the error now and cannot connect.  But usually, after a few minutes, it's okay again.  Same when I connect/disconnect to a VPN.  If there is a fix, it would be most appreciated, especially under Windows...though I guess equally changing this entire plugin so it uses the chat.google.com servers might also help?

**Comment by EionRobb** [comment]
Sorry that this one took so long to track down and fix.  Was having a similar issue in the new Google Chat plugin and fixing it there helped work out how to fix it in the Hangouts plugin

**Comment by rrthomas** [comment]
Well done, and thanks!

**Comment by Martyprod** [comment]
I'm having this issue since 10 mn.. any idea ? I installed the fix (via the Windows installer), but maybe it's not enough ? thx.

**Comment by alexolog** [comment]
Having the issue right now, when travelling.  Connecting to Hangouts via web works, Pidgin throws 429
